src/bun.js/test/jest.zig

```diff
@@ -481,6 +803,49 @@ pub const Expect = struct {
         pub const Set = std.EnumSet(Op);
     };
 
+    pub fn getSnapshotName(this: *Expect, allocator: std.mem.Allocator, hint: string) ![]const u8 {
+        const test_name = this.scope.tests.items[this.test_id].label;
+
+        var length: usize = 0;
+        var curr_scope: ?*DescribeScope = this.scope;
+        while (curr_scope) |scope| {
+            if (scope.label.len > 0) {
+                length += scope.label.len + 1;
+            }
+            curr_scope = scope.parent;
+        }
+        length += test_name.len;
+        if (hint.len > 0) {
+            length += hint.len + 2;
+        }
+
+        var buf = try allocator.alloc(u8, length);
+
+        var index = buf.len;
```

Comment:
instead of incrmeenting an index

lets use a slice directly

```zig
var remain = buf;
        if (hint.len > 0) {
            bun.copy(u8, remain, hint);
           remain = remain[hint..];
```

```

Reply:
it's a little awkward using the slice because it's copied in reverse and the index is decrementing